lib/core_extensions.rb

```diff
@@ -39,9 +39,15 @@ def initialize(*attribute)
 
     def before_destroy(record)
       klasses.each do |klass|
-        record.send(klass.to_sym).each do |what|
-          what = what.to_label unless what.is_a? String
-          record.errors.add :base, _("%{record} is used by %{what}") % { :record => record, :what => what }
+        record.association(klass.to_sym).association_scope.each do |what|
+          unless what.is_a? String
+            if what.class.find_by_id(what.id)
+              what = what.to_label
+            else
+              what = "a hidden %s resource" %  _(what.class.name)
```

Comment:
Yeah, but the "a hidden %s resource" also needs translating.  I think you need to have two full versions of the `%{record} is being used by %{what}` message below - one for the visible and one for the invisible case so it can be translated as one long string.